src/results_analyze/analyzer_plugin.py

```diff
@@ -0,0 +1,133 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/scancode-toolkit/
+# The ScanCode software is licensed under the Apache License version 2.0.
+# Data generated with ScanCode require an acknowledgment.
+# ScanCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with ScanCode or any ScanCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with ScanCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  ScanCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  ScanCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/scancode-toolkit/ for support and download.
+
+import attr
+
+from commoncode.cliutils import PluggableCommandLineOption
+from commoncode.cliutils import POST_SCAN_GROUP
+from plugincode.post_scan import PostScanPlugin
+from plugincode.post_scan import post_scan_impl
+
+from results_analyze import analyzer
+
+
+class NotAnalyzableResourceException(Exception):
+    pass
+
+
+@post_scan_impl
+class ResultsAnalyzer(PostScanPlugin):
+    """
+    Add the "license_detection_errors" list which has the license detection error
+    type information for each match errors.
+    """
+
+    resource_attributes = {
+        "license_detection_analysis": attr.ib(default=attr.Factory(list))
+    }
+
+    sort_order = 80
+
+    options = [
+        PluggableCommandLineOption(
+            ("--analyze-results",),
+            is_flag=True,
+            default=False,
+            help='Add the "license_detection_analysis" list which has the'
```

Comment:
Yes, adding the required options to the help text (also what it does) and updating INSTALL.rst to reflect these required arguments. Thanks++ for this. Will run a scan and paste the output here. 

Also, the `tests/data/analyzer-plugins/results_analyzer_from_sample_json_expected.json` is generated by the `--from-json tests/data/analyzer-plugins/sample_files_result.json --analyze-results` and the `tests/data/analyzer-plugins/results_analyzer_expected.json` is generated by running a scan with `--analyze-results` and the required CLI options on the `tests/data/analyzer-plugins/scan-files/` folder. In case you want to take a look into the outputs and how it looks.

Reply:
Also, running `scancode -li --json-pp sample_license_detection_analysis.json path/to/scancode-toolkit/samples --license-text --is-license-text --classify --analyze-license-results` gave this result:- 
[sample_license_detection_analysis.json.txt](https://github.com/nexB/scancode-results-analyzer/files/5808330/sample_license_detection_analysis.json.txt)

This isn't super interesting as all the license detections are spot on (in only one there's a 99.6 match coverage so it detects and flags that as `near-perfect`). 

Refer to the examples above as they have license detection errors and multiple file-regions in a file.

Reply:
The help text now is:- 
```
help='Performs a license detection analysis to detect different kinds of '
         'potential license detection issues in scancode. '
         'Required scancode CLI options to run this analysis are:'
         '--license --license-text --is-license-text --classify --info'
```